toolkit/tools/pkg/imagecustomizerlib/resolvconf.go

```diff
@@ -0,0 +1,142 @@
+		}
+		existing.existingType = resolvConfTypeSymlink
+		existing.symlinkPath = symlinkPath
+	} else {
+		fileContents, err := file.Read(imageResolveConfPath)
+		if err != nil {
+			return resolvConfInfo{}, fmt.Errorf("failed to read resolv.conf file:\n%w", err)
+		}
+		existing.existingType = resolvConfTypeFile
+		existing.fileContents = fileContents
+	}
+
+	// Remove the existing resolv.conf file, if it exists.
+	// Note: It is assumed that the image will have a process that runs on boot that will override the resolv.conf
+	// file. For example, systemd-resolved. So, it isn't neccessary to make a back-up of the existing file.
+	err = os.RemoveAll(imageResolveConfPath)
+	if err != nil {
+		return resolvConfInfo{}, fmt.Errorf("failed to delete existing resolv.conf file:\n%w", err)
+	}
+
+	err = file.Copy(resolvConfPath, imageResolveConfPath)
+	if err != nil {
+		return resolvConfInfo{}, fmt.Errorf("failed to override resolv.conf file with host's resolv.conf:\n%w", err)
+	}
+
+	return existing, nil
+}
+
+func restoreResolvConf(existing resolvConfInfo, imageChroot *safechroot.Chroot) error {
+	imageResolveConfPath := filepath.Join(imageChroot.RootDir(), resolvConfPath)
+
+	// Delete the overridden resolv.conf file.
+	err := os.RemoveAll(imageResolveConfPath)
+	if err != nil {
+		return fmt.Errorf("failed to delete overridden resolv.conf file: %w", err)
+	}
+
+	switch existing.existingType {
+	case resolvConfTypeNone:
+		// Check if systemd-resolved is enabled.
+		resolvedEnabled := true
+		err := imageChroot.UnsafeRun(func() error {
+			err := shell.ExecuteLive(true, "systemctl", "is-enabled", "systemd-resolved.service")
+
+			var exitError *exec.ExitError
+			if errors.As(err, &exitError) && exitError.ExitCode() == 1 {
+				resolvedEnabled = false
+				return nil
+			}
+			return err
+		})
+		if err != nil {
+			return fmt.Errorf("failed to check if systemd-resolved service is enabled:\n%w", err)
+		}
+
+		if resolvedEnabled {
+			// The systemd-resolved.service is enabled.
+			// So, create the symlink for the resolv.conf file.
+			// While technically this symlink will be automatically created on first boot, doing it now is helpful for
+			// verity images where the root filesystem is readonly.
+			err := os.Symlink(resolvSystemdStubPath, imageResolveConfPath)
+			if err != nil {
+				return fmt.Errorf("failed to create resolv.conf symlink:\n%w", err)
+			}
+		}
+
+	case resolvConfTypeFile:
+		// Restore the resolv.conf file.
+		err := file.WriteWithPerm(existing.fileContents, imageResolveConfPath, 0o644)
+		if err != nil {
+			return fmt.Errorf("failed to restore resolv.conf file:\n%w", err)
+		}
+
+	case resolvConfTypeSymlink:
+		// Restore the resolv.conf symlink.
+		err := os.Symlink(existing.symlinkPath, imageResolveConfPath)
+		if err != nil {
+			return fmt.Errorf("failed to restore resolv.conf file:\n%w", err)
+		}
+
+	default:
+		panic(fmt.Sprintf("unexpected imagecustomizerlib.resolvConfType: %#v", existing.existingType))
```

Comment:
Why not this?
```suggestion
		logger.Log.Panicf("unexpected imagecustomizerlib.resolvConfType: %#v", existing.existingType)
```

Reply:
This was auto-generated by VSCode and I forgot to change it. Fixed.